### DISCUSSION TITLE
Urgently needed functionality

### DISCUSSION THREAD

**Discussion by mieyeaa** [AppFlowy-IO/AppFlowy#3068] [Technical help wanted]
Add Feature1：Insert video function (local video).

Add Feature2：Add sub pages to the page, and then add sub page functions to the sub pages. [Unlimited pages available]

Add Feature3：Code pad inserted like 【notion】

Add Feature4：Visualize the addition of all Feature.

**Comment by Xazin** [comment]
> Add Feature1：Insert video function (local video).
> 
> 
> 
> Add Feature2：Add sub pages to the page, and then add sub page functions to the sub pages. [Unlimited pages available]
> 
> 
> 
> Add Feature3：Code pad inserted like 【notion】
> 
> 
> 
> Add Feature4：Visualize the addition of all Feature.

Hey, thanks for taking an interest in AppFlowy! ☺️

For the first point, we have yet to decide on a way forward for supporting videos on all platforms.

For the second point, we are already working on enabling nested pages. It will be available in the version range 0.2.9 - 0.3.2.

Can you elaborate on point 3 and 4?

**Reply by mieyeaa** [reply]
Similar to 【notion】, there is a page with a window to another page that you can click on to go to.

**Reply by Xazin** [reply]
> Similar to 【notion】, there is a page with a window to another page that you can click on to go to.

Can you show a picture/video? I am still unsure. 👀🙏

**Reply by mieyeaa** [reply]
![00](https://github.com/AppFlowy-IO/AppFlowy/assets/59656688/6c787ab6-d896-4fb9-9faf-a9c7625cd8a2)
like this

**Reply by Xazin** [reply]
> ![00](https://github.com/AppFlowy-IO/AppFlowy/assets/59656688/6c787ab6-d896-4fb9-9faf-a9c7625cd8a2)
> 
> like this

You can insert a link to another page by writing @ and then searching. 👌

**Reply by mieyeaa** [reply]
I want it to be as easy and convenient as 【notion】.

**Reply by annieappflowy** [reply]
We'll continue to improve it. https://github.com/AppFlowy-IO/AppFlowy/issues/2196

**Reply by mieyeaa** [reply]
Nested web pages are implemented, but not like notion, where child pages can be displayed on the parent page, as in the image above

**Reply by mieyeaa** [reply]
Code board function on the page, unable to paste copied code

**Reply by nates-white** [reply]
You can always just use the @ to reference the child page on the parent page

**Reply by mieyeaa** [reply]
This kind of reference is also found in Notation, but it's different from what I'm talking about.

**Reply by mieyeaa** [reply]
I need to add a subpage to the main page, and then the main page shows the subpage, just like notion! It's like adding a folder under a folder, not referencing it from somewhere else.

**Reply by annieappflowy** [reply]
Hi @mieyeaa , you can create a subpage via the `+` icon next to a parent / child page:
<img width="304" alt="image" src="https://github.com/AppFlowy-IO/AppFlowy/assets/12026239/001c4547-c8e9-43d9-a27a-e24b32f6d0e3">
It would be easier to create one through a shortcut. Will work on the improvement. 
Let me know if you need help.

**Reply by mieyeaa** [reply]
But it doesn't show up inside the page, it can only be displayed in the navigation bar, which I expect to create within the page and display within the page. As in "notion".
